src/traces/bar/plot.js

```diff
@@ -51,7 +51,28 @@ function getXY(di, xa, ya, isHorizontal) {
     return isHorizontal ? [s, p] : [p, s];
 }
 
-function plot(gd, plotinfo, cdModule, traceLayer, opts) {
+function transition(selection, opts, makeOnCompleteCallback) {
```

Comment:
Clean wrapper!

It's a little different than what `scatter/plot.js` does, in particular this piece:

https://github.com/plotly/plotly.js/blob/46866fdbb1fdf2e12437a48252f190d07807aa00/src/traces/scatter/plot.js#L67-L73

@antoinerg have you noticed "out of order" transitions?

Your wrapper plays better with [`Lib.makeTraceGroups`](https://github.com/plotly/plotly.js/blob/master/src/lib/make_trace_groups.js), so if there are no side-effects, maybe we could try this pattern in `scatter/plot.js` and maybe fix https://github.com/plotly/plotly.js/issues/3931 while at it.

Reply:
One drawback from your pattern:

- we have to pass `opts` and `makeOnCompleteCallback` around in all the `transition` calls

Reply:
The `scatter/plot.js` uses this trick:

https://github.com/plotly/plotly.js/blob/46866fdbb1fdf2e12437a48252f190d07807aa00/src/traces/scatter/plot.js#L126-L128

Reply:
**This comment is non-blocking**